File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
#include <unordered_map>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_map<int, int> lenMap;

        for (int i = 0; i < nums.size(); i+=1) {
            lenMap[nums[i]] = 0;
        }

        if (lenMap.size() != nums.size()) {
            return true;
        } else {
            return false;
        }

        
    }
};

File: Data Structures & Algorithms/missing-number/submission-4.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = 0
        for i in range(0,len(nums)):
            res |= 1<<nums[i]
        a = (1<<len(nums)+1) - 1
        diff = res^a
        for j in range(0,32):
            t = 1<<j
            if t&diff:
                return j
